docs/code_generation.md

```diff
@@ -36,9 +36,17 @@ There are only two such files currently in use:
 
 ### Installing zap and environment variables
 
-Matter scripts may need to invoke `zap-cli` (for code generation) or `zap` (to
-start the UI tool). For this, scrips need to know where to find the commands. In
-the following order, the scripts process these environment variables:
+ZAP is generally installed as a third-party tool via CIPD during the build
+environment bootstrap (see `scripts/zap.json`), which makes `zap-cli` available
+in `$PATH` when running in a build environment.
+
+**NOTE**: zap packages are currently NOT available for `arm64` (like when
+compiling on Raspberry PI.). In these cases one should checkout zap from source
```

Comment:
Does this part also apply to M1/M2 Mac?  Or does cipd just grab the "mac" version there and use it via Rosetta?

Reply:
I believe we need someone with M1/M2 mac to verify this since I do not actually know.

Reply:
Separate discussion results:
   - M1/M2 does NOT get zap with the current setup
   - looking to add mac to cipd for arm64 as well, since it seems Rosetta makes it work
   
   Expect these changes (use cipd once updated) and documentation to come as a separate PR.